Maple2.Server.Game/PacketHandlers/ItemExtractionHandler.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Maple2.Database.Storage;
+using Maple2.Model.Enum;
+using Maple2.Model.Game;
+using Maple2.Model.Metadata;
+using Maple2.PacketLib.Tools;
+using Maple2.Server.Core.Constants;
+using Maple2.Server.Core.PacketHandlers;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+
+namespace Maple2.Server.Game.PacketHandlers;
+
+public class ItemExtractionHandler : PacketHandler<GameSession> {
+    public override RecvOp OpCode => RecvOp.ItemExtractionScroll;
+
+    #region Autofac Autowired
+    // ReSharper disable MemberCanBePrivate.Global
+    public ItemMetadataStorage ItemMetadata { get; init; } = null!;
+    public TableMetadataStorage TableMetadata { private get; init; } = null!;
+    // ReSharper restore All
+    #endregion
+
+    public override void Handle(GameSession session, IByteReader packet) {
+        long anvilItemUid = packet.ReadLong();
+        long sourceItemUid = packet.ReadLong();
+
+        Item? anvil = session.Item.Inventory.Get(anvilItemUid);
```

Comment:
From L30 to L72, wrap in `lock (session.Item)` to ensure inventory does not change between these checks